src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -0,0 +1,328 @@
+
+  private val minimal = DatasetMetadata(
+    """{
+      |  "titles": ["Lorum ipsum"],
+      |  "descriptions": ["dolor"],
+      |  "dates": [
+      |    { "scheme": "dcterms:W3CDTF", "value": "2018", "qualifier": "dcterms:created" },
+      |    { "scheme": "dcterms:W3CDTF", "value": "2018", "qualifier": "dcterms:available" },
+      |  ],
+      |  "creators": [ { "initials": "B.A.R.", "surname": "Foo" } ],
+      |  "accessRights": { "category": "OPEN_ACCESS" },
+      |  "audiences": [ { "scheme": "", "key": "D35200", "value": ""} ]
+      |}""".stripMargin).getOrElse(fail("parsing minimal json failed"))
+
+  /** provides the verbose namespaces for inline DDM */
+  override val emptyDDM: Elem = DatasetXml(minimal)
+    .doIfFailure{ case e => println(e) }
+    .getOrElse(fail("test preparation failed"))
+    .copy(child = Seq())
+
+  "minimal" should behave like validDatasetMetadata(
+    input = Success(minimal),
+    expectedOutput = Seq(
+      <ddm:profile>
+        <dc:title>Lorum ipsum</dc:title>
+        <dcterms:description>dolor</dcterms:description>
+        <dcx-dai:creatorDetails>
+          <dcx-dai:author>
+            <dcx-dai:initials>B.A.R.</dcx-dai:initials>
+            <dcx-dai:surname>Foo</dcx-dai:surname>
+          </dcx-dai:author>
+        </dcx-dai:creatorDetails>
+        <ddm:created>2018</ddm:created>
+        <ddm:available>2018</ddm:available>
+        <ddm:audience>D35200</ddm:audience>
+        <ddm:accessRights>OPEN_ACCESS</ddm:accessRights>
+      </ddm:profile>,
+      <ddm:dcmiMetadata>
+        <dcterms:dateSubmitted xsi:type="dcterms:W3CDTF">2018-03-22</dcterms:dateSubmitted>
+      </ddm:dcmiMetadata>
+    )
+  )
+
+  "minimal with multiple titles" should behave like validDatasetMetadata(
+    input = Success(minimal.copy(titles = Some(Seq("Lorum", "ipsum")))),
+    subset = { actualDDM => emptyDDM.copy(child = <ddm:profile>{actualDDM \ "profile" \ "title"}</ddm:profile>) },
+    expectedOutput = Seq(
+      <ddm:profile>
+        <dc:title>Lorum</dc:title>
+        <dc:title>ipsum</dc:title>
+      </ddm:profile>
+    )
+  )
+
+  "minimal with rightsHolders" should behave like {
+    val someCreators = Some(Seq(
+      Author(
+        initials = Some("F.O.O."),
+        surname = Some("Bar")
+      ),
+      Author(
+        initials = Some("O."),
+        surname = Some("Belix"),
+        role = Some(SchemedKeyValue("", "RightsHolder", ""))
+      )
+    ))
+    val someContributors = Some(Seq(
+      Author(
+        initials = Some("A.S."),
+        surname = Some("Terix"),
+        role = Some(SchemedKeyValue("", "RightsHolder", ""))
+      )
+    ))
+    validDatasetMetadata(
+      input = Success(minimal.copy(creators = someCreators, contributors = someContributors)),
+      subset = { actualDDM => emptyDDM.copy(child = actualDDM \ "dcmiMetadata") },
+      expectedOutput = Seq(
+        //N.B: creators in ddm:profile unless they are rightsHolders
+        <ddm:dcmiMetadata>
```

Comment:
There are some formatting issues here with your XML.
Note that XML is sometimes not touched by our formatter.
Probably it helps to format it properly here and then add `// @formatter:off` and `// @formatter:on` before and after it respectively.

That formatting thingy is probably not a bad idea for the rest of the XML snippets in the code base as well.

Reply:
Where you say 'somteimes' I experience 'never'

It would be great to have formatting variants. When just XML is selected, format it; if more or nothing is selected, don't format XML.

Reply:
if you find something in the IntelliJ formatter, feel free to do a pull request!